### PR TITLE
more strict 9

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/chat/inbox/row/chat-filter-row/container.tsx [hunk 1]
```diff
@@ -14,45 +14,40 @@ type OwnProps = {
...
-  onBack: dispatchProps.onBack,
-  onEnsureSelection: ownProps.onEnsureSelection,
-  onNewChat: ownProps.onNewChat,
-  onSelectDown: ownProps.onSelectDown,
-  onSelectUp: ownProps.onSelectUp,
-  onSetFilter: ownProps.onQueryChanged,
-  onStartSearch: dispatchProps.onStartSearch,
-  onStopSearch: dispatchProps.onStopSearch,
-})
+const Component = isMobile ? ConversationFilterInput : KeyHandler(ConversationFilterInput)
```
Comment (jakob223) [inline shared/chat/inbox/row/chat-filter-row/container.tsx]: Why do the `isMobile` twice (here and in the definition of `KeyHandler`)? Why put `KeyHandler` in the weird templated function above?

File: shared/common-adapters/avatar-line.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const AvatarLine = (props: Props) => {
   const usernamesToShow = props.usernames.slice(0, props.maxShown)
   const extra = props.usernames.length - usernamesToShow.length
   const reverse = {horizontal: 'horizontalReverse', vertical: 'verticalReverse'} as const
-  const styles = styleMap[props.size][props.layout]
+  const styles = styleMap.get(props.size)![props.layout] ?? {}
```
Comment (jakob223) [inline shared/common-adapters/avatar-line.tsx]: What is this exclamation mark?

File: shared/common-adapters/avatar.tsx [hunk 3]
```diff
@@ -8,7 +8,9 @@ import * as ProfileGen from '../actions/profile-gen'
 import * as Tracker2Constants from '../constants/tracker2'
 import './avatar.css'
 
-export type AvatarSize = 128 | 96 | 64 | 48 | 32 | 24 | 16
+export const avatarSizes = [128, 96, 64, 48, 32, 24, 16] as const
+export type AvatarSize = typeof avatarSizes[number]
```
Comment (jakob223) [inline shared/common-adapters/avatar.tsx]: this is cool

File: shared/common-adapters/avatar-line.tsx (hunk 2 above)
Comment (chrisnojima) [reply shared/common-adapters/avatar-line.tsx]: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-type-assertions

**Pull request merged**